app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def remove_animal(self) -> None:
+        if self in self.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        for _ in self.alive:
+            return (f"{{Name: {self.name}, "
+                    f"Health: {self.health}, "
+                    f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
```

Comment:
Fix annotations for `herbivore` here, plere. If it could be only Herbivore, you wouldn't need the next check in the function